### ISSUE TITLE
File Uploader: UI and validation enhancements

### ISSUE THREAD

**Issue by eporter23** [emory-libraries/dlp-selfdeposit#375]
[Please see this mockup](https://docs.google.com/drawings/d/1XXo2owLQNqO2U0cZJjUoX_JO4hJFxpkkQn2nhdmeyN8/edit?usp=sharing) for notes about how we want to display and validate entries on the Files section of the form.

For this ticket, we want to enhance the visual layout of the Files Tab of the form to be more consistent with the rest of the application interface (see notes in the mockup above).
We also want to enhance usability of the file upload process taking into consideration the following issues:

- [x] Avoid potential issues where if a user has not entered a date or URL in the required format, the form will allow you to save if you proceed to the Files tab and attach a file. [fixed in a separate ticket]
- [x] Disable the "Save" option if a user has not uploaded a file in the Primary Content section. 
- [x] Provide guidance to users for the Attach File process

**Comment by maxdmayhew** [comment]
Giving this an 8. Not sure how difficult this will be to setup. 
May be fairly complex.

**Comment by maxdmayhew** [comment]
1. setup validation how we have been with file upload 
2. styling buttons on file page
3. discussed with Emily and the the above ticket are enough to close this ticket and may need additional work in the future.

**Comment by maxdmayhew** [comment]
There does seem to be a modal bug when trying to save something on the files tab if there is a problem on the descriptions tab

**Comment by maxdmayhew** [comment]
<img width="1267" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMm95Qmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b64cb1053f9aa3e2769f53e74465a80809422ae2/image.png" alt="image.png" />

<img width="1267" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMnd5Qmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d0b05e0c24cacf4c766ae614ebb669d3e569e77b/Screenshot%202024-08-05%20at%201.47.03%E2%80%AFPM.png" alt="Screenshot 2024-08-05 at 1.47.03 PM.png" />

resolved modal tab issue on both new and existing publications

**Comment by eporter23** [comment]
This should be unblocked now because we can attach files manually in the UI again.

**Comment by maxdmayhew** [comment]
improve visual styling as well on this ticket. maybe remove the table is possible

**Comment by maxdmayhew** [comment]
## Article
---
- #### required
	- publisher_version
	- parent_title
- #### hidden fields
	- conference_name
	- isbn
	- series_title
	- edition

## Book
---
- #### required
	- publisher_version
- #### hidden fields
	- parent_title
	- conference_name
	- issn
	- volume
	- issue
	- page_range_start
	- page_range_end

## Book Chapters
---
- #### required
	- publisher_version
	- parent_title
- #### hidden fields
	- conference_name
	- issn
	- volume
	- issue

## Conferences
---
- #### required
	- publisher_version
- #### hidden fields
	- issn
	- isbn
	- series_title
	- edition

## Alway required fields with validation:
- Date is always required now? and has validation.
- Primary Content File

## validation if used:
- Supplemental Material (URL)
- Final Published Version (URL)

**Comment by maxdmayhew** [comment]
I am adding the above really for Emily and anyone who is doing testing as this form has gotten fairly large now. I am a little worried with the field names changing that there may be issues or bugs. I will continue to update this form for my personal notes if further validation work is required.

**Comment by eporter23** [comment]
Thanks @maxdmayhew !

**Comment by maxdmayhew** [comment]
<img width="1145" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeFRQQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0450e130505e003e1ed57d420e320b9258da6018/Screenshot%202024-09-12%20at%2010.52.09%E2%80%AFAM.png" alt="Screenshot 2024-09-12 at 10.52.09 AM.png" />

Can we make this another ticket to style this beyond this point? It might need to be a spike.

**Comment by maxdmayhew** [comment]
Like we spoke about @eporter23 I made it so we are only checking if the file has been added if the user is adding a new publication or "/concern/publications/new". Other wise it will bypass this check.

PR: https://github.com/emory-libraries/dlp-selfdeposit/pull/511

**Comment by eporter23** [comment]
@maxdmayhew looks great so far! One thing I ran into when testing today is that I didn't click the Attach File button after I had selected a file for upload but I was still able to save a brand new work (with no file actually being attached). Let me know if this needs to be a new ticket. I have gotten the modal alert when only adding metadata to the Descriptions tab and then trying to save the work, so this is mostly working.

**Comment by maxdmayhew** [comment]
<img width="610" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMkxnQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0012a7f6546e82dc39c99026ec1dcf0174e0f53f/Screenshot%202024-09-17%20at%203.59.42%E2%80%AFPM.png" alt="Screenshot 2024-09-17 at 3.59.42 PM.png" />

file not selected

<img width="615" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMlBnQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d9d6134273cd69c7cb79f1648a36f7ca51c1ba80/Screenshot%202024-09-17%20at%204.01.17%E2%80%AFPM.png" alt="Screenshot 2024-09-17 at 4.01.17 PM.png" />

file selected but not attached.

@eporter23 Let me know if I should change any of the verbiage and I will submit a PR. Sorry about the bug!

**Comment by eporter23** [comment]
@maxdmayhew no problem! Quick verbiage suggestion: "The Primary Content File must be attached before saving this work. Please review the Files tab of the form and make sure you have selected a file for upload. Then click the Attach File button and make sure the Upload Status is complete before saving the work." (wordy text, but ok for now).

**Comment by maxdmayhew** [comment]
<img width="580" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN25rQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5e72ded0a69dcae4c6c2f3558c14c34fa24f6d14/Screenshot%202024-09-18%20at%2010.34.35%E2%80%AFAM.png" alt="Screenshot 2024-09-18 at 10.34.35 AM.png" />

submitting PR: https://github.com/emory-libraries/dlp-selfdeposit/pull/523

**Comment by maxdmayhew** [comment]
<img width="955" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNUxxQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c97b1d7a93a213efb79d915b15374c5925f6ba1f/Screenshot%202024-09-19%20at%2012.04.57%E2%80%AFPM.png" alt="Screenshot 2024-09-19 at 12.04.57 PM.png" />

This PR also fixed table overflow (due to long pdf title) and changes the button styling. 

<img width="935" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOC9xQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1767eb75a98af075efccd46d12f2c0ba3603fd89/Screenshot%202024-09-19%20at%2012.45.36%E2%80%AFPM.png" alt="Screenshot 2024-09-19 at 12.45.36 PM.png" />

**Comment by eporter23** [comment]
This looks great @maxdmayhew !
